controllers/admin/AdminTranslationsController.php

```diff
@@ -2682,7 +2682,7 @@ public function copyMailFilesForAllLanguages()
                     if (!in_array($file, self::$ignore_folder)) {
                         $files_to_copy_iso[] = array(
                                 "from" => $dir.$file,
-                                "to" => str_replace(_PS_ROOT_DIR_, _PS_ROOT_DIR_.'/themes/'.$current_theme, $dir).$file
+                                "to" => str_replace((strpos($dir, _PS_CORE_DIR_) !== false) ? _PS_CORE_DIR_ : _PS_ROOT_DIR_, _PS_ROOT_DIR_.'/themes/'.$current_theme, $dir).$file
```

Comment:
Even if you use PS_HOSTED_MODE, you have to test for strpos with PS_CORE_DIR because if you didn't have modified your mail translations the files will be in _PS_CORE_DIR_ but if you have modified them they will be in _PS_ROOT_DIR_.
So, I think PS_HOSTED_MODE will make the code more "heavy" for nothing.